mitodl/micromasters
# Added TermsOfServiceDialog and associated stuff

#### What are the relevant tickets?

closes #814 

#### What's this PR do?

This moves agreement to the terms of service to the profile signup flow. Now, there's a check-box on the 'personal' step of the profile flow, and agreeing to the terms of service is required in order to progress beyond the personal step.

This also adds a dialog to show the actually text of the ToS, which can be opened by clicking a link shown next to the checkbox.

#### Where should the reviewer start?

Read over the changes to `PersonalForm.js` and the new `TermsOfServiceDialog.js`.

#### How should this be manually tested?

go to `/profile`.

1. uncheck the ToS checkbox
2. click 'save'
3. confirm that validation errors appear and you cannot move forward.
4. click on 'MIT MicroMasters Terms of Service', confirm that the dialog opens.
5. confirm that clicking 'accept' clears validation errors.

etc?

#### Any background context you want to provide?

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
